cloud_experts_tutorials/rosa-mobb-adding-public-ingress-endpoint-private-link-cluster.adoc

```diff
@@ -0,0 +1,214 @@
+//---
+
+Use this tutorial to create a public ingress endpoint for your {product-title} (ROSA) PrivateLink cluster. 
+
+[IMPORTANT]
+====
+Be aware of the security implications of creating a public subnet in your ROSA virtual private cloud (VPC) with this method.
+====
+
+image::arch-pl-ingress.png[Adding a public ingress endpoint to a ROSA cluster]
+
+For more information on adding public ingresses to a PrivateLink cluster, go link:https://developers.redhat.com/articles/2023/04/27/how-add-public-ingress-private-link-rosa-cluster[here].
+
+.Prerequisites
+
+* The link:https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html[AWS CLI]
+* The link:https://github.com/openshift/rosa/releases/tag/v1.0.8[ROSA CLI] (`rosa`)
+* The `jq` command
+* A link:./README.md[ROSA PrivateLink cluster]
+
+== Getting Started
+
+=== Setting your environment variables
+
+. Set the following environment variables, changing them to suit your cluster:
++
+[source,terminal]
+----
+$   export ROSA_CLUSTER_NAME=private-link
+   # this should be a free CIDR inside your VPC
+$   export PUBLIC_CIDR=10.0.2.0/24
+$   export AWS_PAGER=""
+$   export EMAIL=username.taken@gmail.com
+$   export DOMAIN=public.aws.mobb.ninja
+$   export SCRATCH_DIR=/tmp/scratch
+$   mkdir -p $SCRATCH_DIR
+----
+
+=== Creating a public subnet
+[NOTE]
+====
+If you followed the instructions to create a ROSA PrivateLink cluster linked in the prerequisites, you should already have a public subnet in your VPC. Skip to the last step of this procedure to tag your subnet.
+====
+
+. Obtain a private subnet ID from the cluster:
++
+[source,terminal]
+----
+$   PRIVATE_SUBNET_ID=$(rosa describe cluster -c $ROSA_CLUSTER_NAME -o json \
+      | jq -r '.aws.subnet_ids[0]')
+$   echo $PRIVATE_SUBNET_ID
+----
+
+. Obtain the VPC ID from the subnet ID:
++
+[source,terminal]
+----
+$    VPC_ID=$(aws ec2 describe-subnets --subnet-ids $PRIVATE_SUBNET_ID \
+      --query 'Subnets[0].VpcId' --output text)
+$    echo $VPC_ID
+----
+
+. Obtain the cluster tag from the subnet:
++
+[source,terminal]
+----
+ $  TAG=$(aws ec2 describe-subnets --subnet-ids $PRIVATE_SUBNET_ID \
+      --query 'Subnets[0].Tags[?Value == `shared`]' | jq -r '.[0].Key')
+ $  echo $TAG
+----
+
+. Create a public subnet:
++
+[source,terminal]
+----
+$   PUBLIC_SUBNET=`aws ec2 create-subnet --vpc-id $VPC_ID --cidr-block $PUBLIC_CIDR \
+     --query 'Subnet.SubnetId' --output text`
+$   echo $PUBLIC_SUBNET
+----
+
+. Tag the public subnet for the cluster:
++
+[source,terminal]
+----
+$   aws ec2 create-tags --resources $PUBLIC_SUBNET \
+      --tags Key=Name,Value=$ROSA_CLUSTER_NAME-public \
+      Key=$TAG,Value="shared" Key=kubernetes.io/role/elb,Value=1
+----
+
+=== Creating a custom domain
+
+. Use link:https://certbot.eff.org/[Certbot] to create a Transport Layer Security (TLS) key pair for the custom domain:
++
+[NOTE]
+====
+Skip this step if you already have a key pair.
+====
++
+[source,terminal]
+----
+$   certbot certonly --manual \
```

Comment:
All of these should be lined up under the `$`

Reply:
They are in VS Code but aren't appearing that way in GitHub for some reason.